envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/utils/ReactorUtilsTest.kt

```diff
@@ -73,6 +73,33 @@ class ReactorUtilsTest {
         assertThat(buffered).isEqualTo(2.0)
     }
 
+    @Test
+    fun `should measure discarded items of backpressureLatest`() {
+        // given
+        val meterRegistry = SimpleMeterRegistry()
+
+        // when
+        val received = Flux.range(0, 10)
+            .filter { it % 2 == 0 } // 5 items will be discarded by filter
+            .onBackpressureLatestMeasured("latest", meterRegistry)
+            .subscribeRequestingN(n = 2)
+
+        // then
+        assertThat(received.await(2, TimeUnit.SECONDS)).isTrue()
+
+        val discardedItemsBeforeBackpressure = meterRegistry["reactor-discarded-items.latest-before"].counter().count()
+        val discardedItemsAfterBackpressure = meterRegistry["reactor-discarded-items.latest"].counter().count()
+
+        /**
+         * Published by range: (0..10)
+         * After filtering: (0, 2, 4, 6, 8) (5 discarded)
+         * Requested and dispatched to subscriber: (0, 2)
+         * Not dispatched to subscriber, received by onBackpressure: (4, 6, 8)
+         * Discarded by onBackpressure: (4, 6)
+         */
+        assertThat(discardedItemsAfterBackpressure - discardedItemsBeforeBackpressure).isEqualTo(2.0)
```

Comment:
Can't we assert those values separately rather than calculating difference?

Reply:
We can, but this way we show "how to use" this feature. The difference is important, not absolute values. User should do the same - calculate difference on the dashboard.

I think it's better when a test documents a feature.

Reply:
Ok, it's documented in production code.

Reply:
Since only difference is interesting, couldn't we publish metric from already calculated difference?

Reply:
We can, but it will be slightly harder to implement (we will have to create common context for both doOnDiscard operators and store and increment variables in that context).

I assumed that this is "good enough" and it is not worth to spend more time on it.